### ISSUE TITLE
tf.keras.backend.arange crash (abort) when start is large

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#46889]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A
- 
**Describe the current behavior**
`tf.keras.backend.arange` crash (abort) when `start` is large
**Describe the expected behavior**
expect no crash

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
tf.keras.backend.arange(start=1e+38)
~~~

Output:
~~~python
2021-02-03 16:53:49.181545: F tensorflow/core/framework/tensor_shape.cc:187] Non-OK-status: InitDims(dim_sizes) status: Internal: Expected shape dimensions to be non-negative, got -9223372036854775808
Aborted (core dumped)
~~~

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.4, nightly version(`2.5.0-dev20210203`) and was able to reproduce the issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/4a94ece739a4907b5af0542ca440e6aa/untitled653.ipynb). Thanks!

**Comment by sushreebarsa** [comment]
Was able to reproduce the issue in TF 2.6.0-dev20210528,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/6a229922be4724e77733241763e081fc/untitled42.ipynb)..Thanks !

**Comment by yongtang** [comment]
The issue will be fixed by PR #51711.
